Addgene parser

FixResh opened the pull request:
Script that allows to parse Aggene plasmid repository.

FixResh committed:
Addgene parser

Script that allows to parse Aggene plasmid repository.

romelllo commented on Addgene parser.py:
```diff
@@ -0,0 +1,190 @@
+c = PlasmidParser(id=186478)
+e = PlasmidParser(id=22222)
+
+for plasmid in PlasmidParser.plasmid_list:
```
You can move the for cycle to `main()` function.
Ex: https://realpython.com/python-main-function/
Then, call it in `if __name__ == "__main__":` construction

romelllo commented on Addgene parser.py:
```diff
@@ -0,0 +1,190 @@
+
+
+# Examples
+a = PlasmidParser(id=42888)
```
You can collect all variables for example into a `dict` and give the dict keys meaningful names: a -> pAquaN1 (ex, name of the plasmid).
Ex: PLASMIDS_FOR_TEST = {'pAquaN1': PlasmidParser(id=42888), ...}

romelllo commented on Addgene parser.py:
```diff
@@ -0,0 +1,190 @@
+    def to_csv(self, path):
+        if not os.path.isdir(path + f'Plasmids\\{self.name}'):
+            os.makedirs(path + f'Plasmids\\{self.name}')
+        dct = {k: [v] for k, v in self.__dict__.items()}
```
dct - short version of dict?

FixResh replied:
Yes, I thought it would be easier to read the code with this variable. I think it's possible to get rid of it :)

romelllo commented on Addgene parser.py:
```diff
@@ -0,0 +1,190 @@
+    vector_type: list
+    marker: str
+    resistance: str
+    growth_t: str
```
It is not clear what growth_t (growth_temparature ?) means. You can add a class docstring with an explanation of class attributes.

FixResh replied:
Good point. I'll make more comments about that!

romelllo commented on Addgene parser.py:
```diff
@@ -0,0 +1,190 @@
+    def __str__(self):
+        return f'Plasmid: {self.name}; vendor: {self.vendor}; id: {self.id}; size: {self.size}.'
+
+    def __repr__(self):
```
Built-in method __str__ does the same. Why do we need to specify methods with the same output twice?

FixResh replied:
My idea was about making a shorter version of standart str representation. If this is not making any sense, I'll fix the code then.

romelllo commented on Addgene parser.py:
```diff
@@ -0,0 +1,190 @@
+        return f'Plasmid: {self.name}; vendor: {self.vendor}; id: {self.id}; size: {self.size}.'
+
+    def to_csv(self, path):
+        if not os.path.isdir(path + f'Plasmids\\{self.name}'):
```
Will these paths work on UNIX-systems?

FixResh replied:
These paths work on UNIX-systems, but It seems that there should be a starting script. Anton Guscha append this script to this repository!

romelllo commented on Addgene parser.py:
```diff
@@ -0,0 +1,190 @@
+from bs4 import BeautifulSoup
+import requests
+import pandas as pd
+import urllib.request
```
`import urllib.request as request`

FixResh replied:
It seems reasonable!

romelllo commented on Addgene parser.py:
```diff
@@ -0,0 +1,190 @@
+
+    def to_txt(self, path, doc_seq):
+        sequence = doc_seq.find_all('a', class_='genbank-file-download', href=True)[0]['href']
+        req = urllib.request.Request(sequence, headers={'User-Agent': 'Mozilla/5.0'})
```
`req = request.Request(sequence, headers={'User-Agent': 'Mozilla/5.0'})`
Import change sees above.

FixResh replied:
Changed!

NCrashed committed:
Add script to run on Linux NixOS

FixResh committed:
Addgene parser with fixes

FixResh committed:
Merge branch 'Addgene-parser' of https://github.com/romelllo/plasmids-search-engine into Addgene-parser

FixResh committed:
Addgene parser with fixes

FixResh committed:
Addgene_parser

The parser allows you to get information about plasmids by their ids from Addgene store page.

Commit:
Change request import

Commit:
Deleted old parser

romelllo commented on Addgene_parser.py:
```diff
@@ -0,0 +1,194 @@
+from bs4 import BeautifulSoup
+import requests
+import pandas as pd
+import urllib.request as request
```
Change import to `from urllib.request import request`

romelllo commented on Addgene_parser.py:
```diff
@@ -0,0 +1,194 @@
+
+def main():
+    """A test function that shows how PlasmidParser works"""
+    if __name__ == '__main__':
```
Move from function

FixResh committed:
Fixed import

FixResh committed:
New fixes for parser

romelllo committed:
Merge pull request #3 from NCrashed/Addgene-parser

Add script to run on Linux NixOS

romelllo commented on Addgene_parser.py:
```diff
@@ -0,0 +1,389 @@
+
+main()
+
+"""
```
Before that part there is the same code above. I think the code below is a proper one based on the `main()` part?

FixResh replied:
Done

romelllo reviewed (changes requested):
Part of the code should be deleted. It seems that you forgot to delete the old version.

FixResh committed:
Update Addgene_parser.py

The old copy of the code has been deleted from the file.
